Sources/TensorFlow/Core/LazyTensorTrace.swift

```diff
@@ -89,13 +89,54 @@ class LazyTensorTraceBuilder {
         return materializationTraceInfo([lazyOperation])
     }
 
+    /// Returns a trace obtained by tracing the given function.
+    static func trace<In: TensorGroup, Out: TensorGroup>(_ fn: (In) -> Out) -> LazyTensorTrace {
+        precondition(_RuntimeConfig.useLazyTensor, "Lazy tensor is not enabled for tracing.")
+
+        // Set up inputs for running `fn`.
+        let inputOps = In._typeList.map { Self.makePlaceholder(dataType: $0) }
+        let inputHandles = inputOps.map { LazyTensorHandle(_lazy: $0, index: 0) }
+        let input = In(_handles: inputHandles)
+
+        // Run the function.
+        let output: TensorArrayProtocol = fn(input)
+
+        // Set up the closure that determines if a `LazyTensorOperation` should be an output.
+        let outputLazyOperations = output._tensorHandles.map {
+            (handle: _AnyTensorHandle) -> LazyTensorOperation in
+            let lazyOp = lazyTensorOperation(handle)
+            precondition(lazyOp != nil, "Found a non-lazy tensor in output when tracing.")
+            return lazyOp!
+        }
+        let outputIDs = Set<ObjectIdentifier>(outputLazyOperations.map { ObjectIdentifier($0) })
```

Comment:
Use `.lazy` to avoid creating a temporary array.
```suggestion
        let outputIDs = Set<ObjectIdentifier>(outputLazyOperations.lazy.map { ObjectIdentifier($0) })
```